pr-request-template.md

```diff
@@ -0,0 +1,77 @@
+# Pull Request Template
+
+Thank you for your contribution to Alumnium!
+Please fill out this template completely to help us understand and review your PR efficiently.
+
+## Summary
+What is this pull request about?
+Provide a brief description of what the PR does, the problem it solves, and any context around the implementation.
+
+## Motivation
+Why is this change being made?
+Describe the use case, problem, or opportunity that this PR addresses.
+Explain how this change contributes to the project's goals or improves functionality.
+
+## Related Issue(s)
+If this PR fixes or is related to an open issue, link it here.
+Example: Closes #123
+
+Closes #
+
+## Type of Change
+Mark the relevant options.
+
+- [ ] Bug fix  
+- [ ] New feature  
+- [ ] Documentation update  
+- [ ] Test coverage improvement  
+- [ ] Refactoring  
+- [ ] CI/CD enhancement
+- [ ] Breaking change
+- [ ] Other (please specify):
+
+## Code or UI Demos (if applicable)
+Add screenshots, logs, terminal output, or code snippets that help reviewers understand the change.
+Useful especially for UI changes or error fixes.
+
+## Testing Strategy
+How have you tested your changes?
+Mention tools used, test cases added or updated, and how the change was validated.
+
+## Local Testing
+Describe the environment and steps you used to test these changes locally.
+Include any relevant configuration details or test data.
```

Comment:
@automatealchemist, what is the idea for these sections? You added a checklist below to remind contributors to test locally and add tests. We can skip these, as all the test changes will be visible in the file changes, and it feels redundant to ask to list the updated tests.